content/chapitres/ci.adoc

```diff
@@ -30,56 +30,98 @@ Source : http://cartoontester.blogspot.be/2010/01/big-bugs.html
+
+== Concepts de GitHub Actions - Step 2/2
 
 Une *Step* (étape) est une tâche individuelle à faire effectuer par le CI :
 
 * Par défaut c'est une commande à exécuter - mot clef `run`
 * Ou une "action" (quel est le nom du produit déjà ?) - mot clef `uses`
-** Réutilisables et https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/finding-and-customizing-actions[partageables]
+** Réutilisables et https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/finding-and-customizing-actions[partageables,window=_blank]
 
-== Concepts de Github Actions - Job
+[source,yaml]
+----
+steps: # Liste de steps
+  # Exemple de step 1 (commande)
+  - name: Say Hello
+    run: echo "Hello ENSG"
+  # Exemple de step 2 (une action)
+  - name: 'Login to DockerHub'
+    uses: docker/login-action@v1 # https://github.com/marketplace/actions/docker-login
+    with:
+      username: ${{ secrets.DOCKERHUB_USERNAME }}
+      password: ${{ secrets.DOCKERHUB_TOKEN }}
+----
+
+== Concepts de GitHub Actions - Job 1/2
+
+image::gh-actions-concepts.jobs.svg[width=800]
+
+== Concepts de GitHub Actions - Job 2/2
 
 Un *Job* est un groupe logique de tâches :
 
 * Enchaînement _séquentiel_ de tâches
-* Regroupement logique : "qui a un sens" (exemple : )
+* Regroupement logique : "qui a un sens"
+** Exemple : "compiler puis tester le résultat de la compilation"
 
-== Concepts de Github Actions - Runner
+[source,yaml]
+----
+jobs: # Map de jobs
+  build: # 1er job, identifié comme 'build'
+    name: 'Build Slides'
+    runs-on: ubuntu-latest # cf. prochaine slide "Concepts de GitHub Actions - Runner"
+    steps: # Collection de steps du job
+      - name: 'Build the JAR'
+        run: mvn package
+      - name: 'Run Tests on the JAR file'
+        run: mvn verify
+  deploy: # 2nd job, identifié comme 'deploy'
+    # ...
+----
+
+== Concepts de GitHub Actions - Runner
 
 Un *Runner* est un serveur distant sur lequel s'exécute un job.
 
 * Mot clef `runs-on` dans la définition d'un job
-* Défaut : machine virtuelle Ubuntu dans le cloud utilisé par Github
-* https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners[D'autres types sont disponibles]
+* Défaut : machine virtuelle Ubuntu dans le cloud utilisé par GitHub
+* https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners[D'autres types sont disponibles,window=_blank]
 (macOS, Windows, etc.)
-* Possibilité de fournir https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners[son propre serveur]
+* Possibilité de fournir https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners[son propre serveur,window=_blank]
+
+== Concepts de GitHub Actions - Workflow 1/2
 
-== Concepts de Github Actions - Workflow
+image::gh-actions-concepts.workflows.svg[width=800]
+
+== Concepts de GitHub Actions - Workflow 2/2
```

Comment:
I would repeat somewhere that:
-  jobs in a workflow are executed in the most parallelized way,
- can depend on each other. We're relying on it also in the next stages.
- Can be conditionally executed, just like steps.